azkaban-common/src/main/java/azkaban/project/AzkabanNode.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.project;
+
+import static java.util.Objects.requireNonNull;
+
+import azkaban.utils.Props;
+
+public abstract class AzkabanNode {
```

Comment:
The class is `abstract` by design. The idea is that a DAG `Node` can be either a `Job` or an embedded `Flow` or may be something else. Here job and flow are concrete classes but a node technically doesn't mean anything. Hence it is an abstract class.

##### Why is it not an interface?
It could be! Just that I think there might be some common code between a Job and a Flow which we can house here.